sdk/daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/NodeHashSpec.scala

```diff
@@ -0,0 +1,1171 @@
+                                     |# Package Name
+                                     |0000000e - [14 (int)]
+                                     |7061636b6167652d6e616d652d30 - [package-name-0 (string)]
+                                     |# Template Id
+                                     |00000007 - [7 (int)]
+                                     |7061636b616765 - [package (string)]
+                                     |00000001 - [1 (int)]
+                                     |00000006 - [6 (int)]
+                                     |6d6f64756c65 - [module (string)]
+                                     |00000001 - [1 (int)]
+                                     |00000004 - [4 (int)]
+                                     |6e616d65 - [name (string)]
+                                     |# Arg
+                                     |00000005 - [5 (int)]
+                                     |68656c6c6f - [hello (string)]
+                                     |# Signatories
+                                     |00000002 - [2 (int)]
+                                     |00000005 - [5 (int)]
+                                     |616c696365 - [alice (string)]
+                                     |00000003 - [3 (int)]
+                                     |626f62 - [bob (string)]
+                                     |# Stakeholders
+                                     |00000002 - [2 (int)]
+                                     |00000005 - [5 (int)]
+                                     |616c696365 - [alice (string)]
+                                     |00000007 - [7 (int)]
+                                     |636861726c6965 - [charlie (string)]
+                                     |# Global Keys
+                                     |00000001 - [1 (int)]
+                                     |# Maintainers
+                                     |00000001 - [1 (int)]
+                                     |00000005 - [5 (int)]
+                                     |6461766964 - [david (string)]
+                                     |# Template Id
+                                     |00000001 - [1 (int)]
+                                     |0000000a - [10 (int)]
+                                     |6d6f64756c655f6b6579 - [module_key (string)]
+                                     |00000001 - [1 (int)]
+                                     |00000004 - [4 (int)]
+                                     |6e616d65 - [name (string)]
+                                     |# Package Name
+                                     |00000010 - [16 (int)]
+                                     |7061636b6167655f6e616d655f6b6579 - [package_name_key (string)]
+                                     |# Key
+                                     |00000005 - [5 (int)]
+                                     |68656c6c6f - [hello (string)]""".stripMargin
```

Comment:
This is maybe a bit extreme, but my goal is to be able to point external partners to this file and hope that in combination with the spec (in the process of writing it now), they can self diagnose most issues.
And it could also be run on arbitrary transactions to help with debugging of specific use cases.